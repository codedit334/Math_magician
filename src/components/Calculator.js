import React from 'react';
import CalcInput from './CalcInput';
import CalcKeys from './CalcKeys';
import '../styles/calculator.css';

export default function Calculator() {
  return (
    <div className="calculator">
      <CalcInput />
      <CalcKeys />
    </div>
  );
}
