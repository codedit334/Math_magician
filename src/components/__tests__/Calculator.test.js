import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

test('Sould render Calculator Component', () => {
  render(<Calculator />);
  const CalculatorElement = screen.getByTestId('Calculator');
  expect(CalculatorElement).toBeInTheDocument();

  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
