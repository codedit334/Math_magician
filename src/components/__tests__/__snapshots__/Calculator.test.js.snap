// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sould render Calculator Component 1`] = `
<div
  className="calculator-page"
>
  <h1>
    Let's do some math!
  </h1>
  <div
    className="calculator"
    data-testid="Calculator"
  >
    <div
      className="calcinput"
    >
      <input
        className="calcinput-display"
        data-testid="CalcInput"
        disabled={true}
        placeholder={0}
        value=""
      />
    </div>
    <div
      className="calcKeys"
    >
      <table>
        <tbody
          data-testid="CalcKeysTBody"
        />
      </table>
    </div>
  </div>
</div>
`;
